controllers/subscriptions.js

```diff
@@ -225,28 +214,35 @@ exports.addEmailPOST = async ( req, res, next ) => {
 		res.redirect( _errorPage );
 		return true;
 	}
-	
-	// Get the topic
-	const topic = await getTopic( topicId );
-	
+
 	try {
-		
-		// No topic = no good
-		if ( !topic || !topic.inputErrURL || !topic.thankURL || !topic.failURL ) {
-			console.log( "addEmailPOST: no topic" );
-			res.redirect( _errorPage );
-			return true;
-		}
-		
 		// Validate if email is the good format (something@something.tld)
-		if ( !email.match( /.+\@.+\..+/ ) ) {
-			res.redirect( topic.inputErrURL );
+		if ( !email || typeof email !== "string" || !email.match( /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/ ) ) {
+			console.log( "addEmail: bad email: " + email );
+			res.json( _cErrorsJSO );
 			return;
 		}
-		
+
+		email = email.toLowerCase() || "";
 		// URL Decrypt the email. It is double encoded like: "&amp;#39;" should be "&#39;" which should be "'"
 		email = await entities.decodeXML( await entities.decodeXML( email ) );
 
+		if ( !topicId ) {
```

Comment:
You'll have to switch the ordering of the topic checks and the email checks, because the email checks would redirect users to the "input error page" defined by the topic as proposed at line 222.